modules/ref/pages/client-settings.adoc

```diff
@@ -96,524 +123,523 @@ By default the client will use a simple matching heuristic to determine which se
+The default is quite high because some operations (such as flushing a bucket, for example) might take a long time.
 
-////
-Name: *Connect Timeout*::
-Parameter: `TimeoutConfig.connectTimeout(Duration)`
+Name: *DNS SRV Timeout*::
+Options Class: `ClusterTimeoutOptions`
 +
-Default: `10s`
+Parameter: `dns_srv_timeout: Optional[timedelta]`
 +
-System Property: `com.couchbase.env.timeout.connectTimeout`
+Default:  `None`
 +
-The connect timeout is used when a Bucket is opened and if not overridden by a custom timeout.
-If you feel the urge to change this value to something higher, there is a good chance that your network is not properly set up.
-Connecting to the server should in practice not take longer than a second on a reasonably fast network.
+// TODO: better description
+DNS SRV timeout.
 
-Name: *Disconnect Timeout*::
-Parameter: `TimeoutConfig.disconnectTimeout(Duration)`
+// TODO:  Python SDK - move to ClusterOptions?
+Name: *Config Idle Redial Timeout*::
+Options Class: `ClusterTimeoutOptions`
 +
-Default: `10s`
+Parameter: `config_idle_redial_timeout: Optional[timedelta]`
 +
-System Property: `com.couchbase.env.timeout.disconnectTimeout`
+Default:  `None`
 +
-The disconnect timeout is used when a Cluster is disconnected and if not overridden by a custom timeout.
-A timeout is applied here always to make sure that your code does not get stuck at shutdown.
-The default should provide enough time to drain all outstanding operations properly, but make sure to adapt this timeout to fit your application requirements.
+// TODO: better description
+Config Idle Redial Timeout.
 
-Name: *Management Timeout*::
-Parameter: `TimeoutConfig.managementTimeout(Duration)`
+=== Compression Options
+
+Name: *Enabling Compression*::
+Options Class: `ClusterOptions`
 +
-Default: `75s`
+Parameter: `enable_compression: Optional[bool]`
 +
-System Property: `com.couchbase.env.timeout.managementTimeout`
+Default:  `True`
 +
-The management timeout is used on all cluster management APIs (BucketManager, UserManager, CollectionManager, QueryIndexManager, etc.) if not overridden by a custom timeout.
-The default is quite high because some operations (such as flushing a bucket, for example) might take a long time.
+If enabled, the client will compress documents before they are sent to Couchbase Server.
+// If this is set to `False`, the other CompressionExample settings have no effect.
+// TODO:  compression_min_size, compression_min_ratio
 
-== Diagnostics Options
+=== Tracing Options
 
-The client can periodically log diagnostic information about server connectivity.
+// TODO:  all options need better descriptions + default values
+Tracing settings are configured using an instance of `ClusterTracingOptions` or passed in as keyword arguments.
 
-Diagnostics settings are represented by the Java class `DiagnosticsConfig`.
-The associated `ClusterEnvironement.Builder` method is called `diagnosticsConfig`.
+.Creating a new `ClusterTracingOptions` instance:
```

Comment:
The example below *doesn't* create a ClusterTractingOptions instance does it?

That is, it seems to use the "passed in as keyword arguments" method instead?

Reply:
Oh, in fact it creates a ClusterTimeoutOptions, is that just a typo?

Reply:
oops -- that is a typo, should be a ClusterTracingOptions() instance.